src/Common/src/TypeSystem/Common/TypeSystemException.cs

```diff
@@ -193,6 +193,10 @@ public static string OwningModule(MethodDesc method)
 
             public static string OwningModule(TypeDesc type)
             {
+#if TYPE_LOADER_IMPLEMENTATION
```

Comment:
Regarding poisoned strings, I don't think it would be a good idea to implement that on name and namespace getters for nometadata types. If we don't have metadata, we already have an unfortunate existing behavior (ex: in MissingMetadata exceptions) where we print the EEType's rva or pointer. We don't really have a good solution there, but I don't think we should add more poisoned names to the system.